app/src/main/java/com/duckduckgo/app/cta/ui/Cta.kt

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (c) 2019 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.cta.ui
+
+import android.net.Uri
+import android.view.View
+import androidx.annotation.AnyRes
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import androidx.fragment.app.FragmentActivity
+import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.cta.model.CtaId
+import com.duckduckgo.app.global.baseHost
+import com.duckduckgo.app.global.install.AppInstallStore
+import com.duckduckgo.app.global.install.daysInstalled
+import com.duckduckgo.app.global.view.DaxDialog
+import com.duckduckgo.app.global.view.hide
+import com.duckduckgo.app.global.view.html
+import com.duckduckgo.app.global.view.show
+import com.duckduckgo.app.onboarding.store.OnboardingStore
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.app.trackerdetection.model.TrackingEvent
+import kotlinx.android.synthetic.main.include_cta_buttons.view.*
+import kotlinx.android.synthetic.main.include_cta_content.view.*
+import kotlinx.android.synthetic.main.include_dax_dialog_cta.view.dialogTextCta
+import kotlinx.android.synthetic.main.include_dax_dialog_cta.view.hiddenTextCta
+import kotlinx.android.synthetic.main.include_dax_dialog_cta.view.primaryCta
+
+interface Cta {
+    val ctaId: CtaId
+    val shownPixel: Pixel.PixelName?
+    val okPixel: Pixel.PixelName?
+    val cancelPixel: Pixel.PixelName?
+
+    fun pixelShownParameters(): Map<String, String?>
+    fun pixelCancelParameters(): Map<String, String?>
+    fun pixelOkParameters(): Map<String, String?>
+
+    fun apply(view: View)
+    fun createDialogCta(activity: FragmentActivity): DaxDialog?
```

Comment:
You asked for some input here and I agree that it is confusing that some CTAs known about DaxDialog, a specific type closely coupled with some CTA types. The solution may be to delete this method and just create these in the caller (could use a factory to keep the code out of those classes) or make this a creation pattern we use across the board but with an interface rather than a concrete type.  It needs some thought so let's add it o the tidy up tasks that we do before rolling this out to everyone.

Reply:
Thanks, I have created https://app.asana.com/0/1125189844152671/1154748750174828 for this.